source/numeric/SVIntHelpers.h

```diff
@@ -108,6 +108,53 @@ static void signExtendCopy(uint64_t* output, const uint64_t* input, bitwidth_t o
     memset(output + oldWords, isNegative ? -1 : 0, (newWords - oldWords) * sizeof(uint64_t));
 }
 
+// Check whether all the bits above word.bit is identical to word.bit.
+static bool isSignExtended(const uint64_t* input, uint32_t numWords, uint32_t word, uint32_t bit,
+                           uint64_t topWordMask) {
+    bool sign = (input[word] & (1ULL << bit)) != 0;
+    if (sign && numWords - 1 > word) {
+        if (input[numWords - 1] != topWordMask)
+            return false;
+        numWords--;
+        topWordMask = UINT64_MAX;
+    }
+
+    uint64_t mask = sign ? UINT64_MAX : 0;
+    for (auto i = numWords - 1; i > word; i--) {
+        if (input[i] != mask)
+            return false;
+    }
+
+    if (numWords - 1 == word)
+        mask &= topWordMask;
+    mask >>= bit;
+    return (input[word] >> bit) == mask;
+}
+
+// If word.bit is one, make all bits above word.bit become one.
```

Comment:
"Otherwise, make them all zero."

Reply:
I make the change.

Reply:
Instead of changing the comment, I change the implementation to match the comment. When the function is called by SVInt::ashr, all bits above msb must be zero. Not doing redundant zero copy improves efficiency a little bit. The comment for SVInt::signExtendFrom is changed to this updated implementation:
```c++
    /// If bit msb is nonzero, duplicate it to all bits in [bitWidth-1:msb].
    void signExtendFrom(bitwidth_t msb);
```